foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -269,6 +269,14 @@
     protected Boolean useJDBCStoredProcedureSyntax;
     protected String driverName;
 
+    /** Table existence check trigger.
+     *  Returned result set is checked on not being empty when {@code true}
+     *  or {@code true} value is always returned when no exception is thrown when {@code false}. */
+    protected boolean shouldCheckResultTableExistsQuery;
```

Comment:
I see what you are trying to do here with the boolean. We want to be able to address both prospects of "try/fail" & "check result length". But I think we should leave that detail (the "how do we actually determine the table exists" detail) up to the individual platform classes. I think it really is as straight-forward as having just one method `public boolean checkTableExists(final DatabaseSessionImpl session, final TableDefinition table)` that the individual platform classes can override. I understand the consequence of code duplication, but I think the amount of copy/paste is negligible:

DatabasePlatform:
``` java
public boolean checkTableExists(final DatabaseSessionImpl session, final TableDefinition table) {
    final DataReadQuery query = new DataReadQuery("SELECT * FROM " + table.getFullName());
    try {
        session.executeQuery(query);
    } catch (Exception notFound) {
        return false;
    }
    return true;
}
```
OraclePlatform:

``` java
public boolean checkTableExists(final DatabaseSessionImpl session, final TableDefinition table) {
    final DataReadQuery query = new DataReadQuery("SELECT table_name FROM user_tables WHERE table_name='" + table.getFullName() + "'");
    query.setMaxRows(1);
    final Vector<?> result = new Vector<>();
    try {
        result = (Vector)session.executeQuery(query);
    } catch (Exception notFound) {
        return false;
    }
    return !result.isEmpty();
}
```

Reusing code is certainly admirable! But I'd personally prefer a simpler design that boasts readability/maintainability and is less complex :/

Reply:
@dazey3: I see your point. The problem is, that each of us (me, Lukas and you) have bit different opinion about how to implement this. :) I would use that interface from previous PR which is from Java point of view optimal solution. Lukas prefers this version with if and you what you proposed.
Anyway, current code is result of compromise and resulting code is still quite simple. I would like to keep it. If overriding class needs its own way, it can still override both methods.

Reply:
Well, I still disagree that this boolean implementation is clean, but if I am outvoted 2:1, that's fine by me. I don't argue this code most likely works, just that it seems a bit complex for what I saw as an otherwise simple task. We have lots of boolean flags in the code all over the place. I'd personally rather limit creating more. I'll approve assuming Lukas agrees.

Reply:
@dazey3, @lukasj: Well, I personally don't much care whether this will be there or not. It's another feature Lukas asked me to add. So you two make some agreement and let me know the result. :)

Reply:
Well, each of the version has some good and bad things:
 * Interface (me):
   - -More complex code around the interface, worse human readibility
   - -Platform code contains class definition
   - +No copy-pastes
   - +Proper java pattern, allows lambdas to implement platform specific things
 * Boolean flag (Lukas):
   - -Additional boolean flag and condition around return statement
   - +Still good human readibility
   - +No copy-pastes
 * Pure method overrides (Will):
   - -Each overriding method copy-pastes query result evaluation code change
   - +Probably best human readibility

I already gave up my solution so you two should make an agreement which way to go. :)